### ISSUE TITLE
UsdShade Python Wrapping Crash

### ISSUE THREAD

**Issue by superfunc** [PixarAnimationStudios/USD#547]
### Description of Issue
Running the following code causes a crash in the python binding for UsdShadeOutput

### Steps to Reproduce
```
>>> from pxr import UsdShade, Usd, Glf
>>> stage = Usd.Stage.CreateInMemory()
>>> p = UsdShade.Material.Define(stage, '/test')
>>> o = p.CreateSurfaceOutput('glslfx')
>>> o.ConnectToSource(p, UsdShade.Tokens.outputsSurface)
fish: 'python' terminated by signal SIGSEGV
``` 

### System Information (OS, Hardware)
Seen on both RHEL and MacOS 

### Package Versions
Standard VFX Platform on RHEL
build_usd.py on MacOS

### Build Flags
Default

**Comment by spiffmon** [comment]
Thanks, @superfunc - I can repro locally.  It seems to be a problem in the
conversion from Material to ConnectableAPI for the first parameter - a
problem that is not shared by the Shader to  ConnectableAPI conversion.
Relatedly, just wanted to point out:

* The connection you're trying to make normally *would be* to an output on
a Shader.  e.g.:
>>> s = UsdShade.Shader.Define(stage, '/test/shader')
>>> o.ConnectToSource(s, UsdShade.Tokens.outputsSurface)

* The token you want to be using here is UsdShade.Tokens.surface, *not*
outputsSurface... we should see if we actually need outputsSurface to be a
public token, as it is definitely confusing.

* You can work around the crash problem for now with an explicit conversion
to ConnectableAPI:
 >>> o.ConnectToSource(UsdShade.ConnectableAPI(p), UsdShade.Tokens.surface)

--spiff

**Comment by superfunc** [comment]
Thanks for explaining the connection bits. I saw the workaround and forgot to post it, so thanks for adding that!

**Comment by jtran56** [comment]
Filed as internal issue #162470.

**Comment by sunyab** [comment]
This issue was addressed with commit 62c90a99 released in v18.09. Closing this out, thanks!
